clients/include/testing_gemv_strided_batched.hpp

```diff
@@ -0,0 +1,442 @@
+    rocblas_int       incx        = arg.incx;
+    rocblas_int       incy        = arg.incy;
+    T                 h_alpha     = static_cast<T>(arg.alpha);
+    T                 h_beta      = rocblas_isnan(arg.beta) ? 0 : static_cast<T>(arg.beta);
+    rocblas_operation transA      = char2rocblas_operation(arg.transA);
+    rocblas_int       stride_a    = arg.stride_a;
+    rocblas_int       stride_x    = arg.stride_x;
+    rocblas_int       stride_y    = arg.stride_y;
+    rocblas_int       batch_count = arg.batch_count;
+
+    rocblas_local_handle handle;
+    size_t               size_A = lda * static_cast<size_t>(N);
+    size_t               size_x, dim_x, abs_incx;
+    size_t               size_y, dim_y, abs_incy;
+
+    if(transA == rocblas_operation_none)
+    {
+        dim_x = N;
+        dim_y = M;
+    }
+    else
+    {
+        dim_x = M;
+        dim_y = N;
+    }
+
+    abs_incx = incx >= 0 ? incx : -incx;
+    abs_incy = incy >= 0 ? incy : -incy;
+
+    size_x = dim_x * abs_incx;
+    size_y = dim_y * abs_incy;
+
+    // argument sanity check before allocating invalid memory
+    if(M < 0 || N < 0 || lda < M || lda < 1 || !incx || !incy || stride_a < size_A
+       || stride_x < size_x || stride_y < size_y || batch_count < 0)
+    {
+        static const size_t safe_size = 100; // arbitrarily set to 100
+        device_vector<T>    dA1(safe_size);
+        device_vector<T>    dx1(safe_size);
+        device_vector<T>    dy1(safe_size);
+        if(!dA1 || !dx1 || !dy1)
+        {
+            CHECK_HIP_ERROR(hipErrorOutOfMemory);
+            return;
+        }
+
+        EXPECT_ROCBLAS_STATUS(rocblas_gemv_strided_batched<T>(handle,
+                                                              transA,
+                                                              M,
+                                                              N,
+                                                              &h_alpha,
+                                                              dA1,
+                                                              lda,
+                                                              stride_a,
+                                                              dx1,
+                                                              incx,
+                                                              stride_x,
+                                                              &h_beta,
+                                                              dy1,
+                                                              incy,
+                                                              stride_y,
+                                                              batch_count),
+                              rocblas_status_invalid_size);
+
+        return;
+    }
+
+    //quick return
+    if(!M || !N || !batch_count)
+        return;
+
+    size_A = size_A + static_cast<size_t>(stride_a) * static_cast<size_t>(batch_count - 1);
```

Comment:
This calculation is basically `size = (size of data) + (size of empty space between batches)`, correct?
If so, shouldn't the calculation be, for example:
`size_A = size_A + (stride_a - size_A) * (batch_count - 1)` ? Or am I misinterpreting what is being done here?

Reply:
For casting, I prefer the syntax:
```
size_A += size_t(stride_a - size_A) * (batch_count -1);
```
or:
```
size_A += (stride_a - size_A) * size_t(batch_count -1);
```

Reply:
No, this is not quite right. What is being calculated here is the size of the entire array to be allocated. This size is essentially: 
`stride_a*batch_count`
When `stride_a < size_A`, one may want to save on the last `stride_a - size_A` locations, and allocate only: 
`stride_a*batch_count - (stride_a - size_A)`  
 which leads to:
`size_A + stride_a*(batch_count - 1)`